### PR TITLE
Set up Grafana using CloudFormation

### PR DESCRIPTION
**Pull request by kathy-t**
For [SEAB-932](https://ucsc-cgl.atlassian.net/browse/SEAB-932)

automate_cli_metrics.sh
- Modified automate_cli_metrics.sh to check for empty arrays because it wasn't working on the dev Grafana VM due to an unbound variable error that was caused by an empty array. I didn't see this error when running it in the Grafana VM in oicr-dockstore because that VM had bash version 4.4, which doesn't throw an unbound variable error for empty arrays. Versions < 4.4 throw this error and the dev Grafana VM has version 4.3.
- Made the metric filter name and metric name the same so it's easier to search

log-group.yml
- Added CloudWatch metrics so they can be created using CloudFormation
- Added refresh by language metrics

dockstore.yml
- Used configsets for DockstoreInstance and DashboardInstance in preparation for adding CW Agent
- Added automate_cli_metrics.sh to DashboardInstance and set up a cron job for it
- Added dashboard and datasource provisioning for Grafana so CloudWatch is added as a datasource and a dashboard is ready upon startup.

For now, I created an S3 bucket to store the Grafana dashboard json file so the dashboard is able to be setup using CloudFormation.

### CODE REVIEW

File: dockstore.yml [hunk 1]
```diff
@@ -601,34 +632,132 @@ Resources:
...
+          files:
+            "/home/ubuntu/automate_cli_metrics.sh":
+              source: "https://raw.githubusercontent.com/dockstore/dockstore-deploy/develop/automate_cli_metrics.sh"
+              mode: "774"
+              owner: ubuntu
+              group: ubuntu
+            "/tmp/dashboardCronTab":
+              # 1. Empty line in the content on purpose; some crontab require a new line at the end
+              # 2. 6 is 3am eastern, midnight pacific
+              # 3. Probably not the right place for the logs
```
Comment (denis-yuen) [inline dockstore.yml]: probably, a common place for logs is `/var/log/dashboard.cron.log`
Another reasonable place would be `/home/ubuntu/logs/dashboard.cron.log`

File: log-group.yml [hunk 2]
```diff
@@ -12,3 +12,254 @@ Resources:
       Properties:
         LogGroupName: !Ref LogGroupName
 
+  # Metric filters for the log group created. These will be displayed as CloudWatch metrics.
+  AnVILRefererCount:
```
Comment (denis-yuen) [inline log-group.yml]: Not much to add, except this file is pretty cool

File: dockstore.yml [hunk 3]
```diff
@@ -371,13 +374,36 @@ Resources:
...
+        Version: '2008-10-17'
+        Id: MyPolicy
+        Statement:
+          - Sid: ReadAccess
+            Action:
+              - s3:GetObject
+            Effect: Allow
+            Resource:
+              - 'arn:aws:s3:::grafana-dashboard-json/*'
+            Principal:
```
[the remark this answers is not in the document]
Comment (kathy-t) [reply dockstore.yml]: Yeah, that's correct. This bucket policy for ``grafana-dashboard-json`` grants read-only access to the cloudformation user

File: dockstore.yml [hunk 4]
```diff
@@ -371,13 +374,36 @@ Resources:
...
+      Bucket: 'grafana-dashboard-json'
 
   DockstoreInstance:
     Type: AWS::EC2::Instance
     Metadata: 
-      AWS::CloudFormation::Init: 
-        config: 
-          packages: 
+      AWS::CloudFormation::Init:
+        configSets:
```
Comment (coverbeck) [inline dockstore.yml]: 👍 I'd been meaning to look into configSets and had never gotten around to it.

**Pull request merged**